tox.ini

```diff
@@ -91,6 +91,13 @@ commands =
     --profile=prospector \
     --die-on-tool-error {posargs}
 
+[testenv:pre-commit]
+description = run pre-commit to ensure the changes are valid against our lintings
+deps =
+   pre-commit
+commands =
+    pre-commit run
```

Comment:
I'm in doubt here about how `darker` will find the diff to check since there won't be files to check at this point because everything is already committed.

We may need to specify `PRE_COMMIT_FROM_REF` (to the base branch) and `PRE_COMMIT_TO_REF` (to latest commit of this branch) to work around this.
See https://github.com/akaihola/darker#customizing-darker-black-and-isort-behavior

Reply:
https://github.com/akaihola/darker/issues/260

Reply:
Somehow, we have to pass `github.event.pull_request.base.sha` and `github.event.pull_request.head.sha` to CircleCI to pass this values to `darker` as pre-commit environment variables.

Reply:
CircleCI built in env vars might give you at least one of those. CIRCLE_SHA1 is available, but the base sha might not be